### PR TITLE
Management command to export discussion participation statistics per course

### PR DESCRIPTION
**Pull request by e-kolpakov**
Background: This PR adds management command to export discussion participation statistics per course.
JIRA ticket: OC-550
Dependencies: Partial dependency on https://github.com/edx-solutions/cs_comments_service/pull/13 - without it should work fine, except filtering by thread type.
Partner information: 3rd party-hosted open edX instance, for an edX solutions client.
Merge deadline: Monday, 2015-03-09.

Author concerns: there seems to be no unit tests for `lms/lib/comment_client/user.py` and for management commands in general. Due to tight deadline, this parts were left with no additional test coverage, but tested manually. Hints on how to best test them are welcome.

Testing instructions: issuing the following command as `edxapp` user should trigger export process with described results:

* `./manage.py lms export_discussion_participation <course_key>` - exports entire discussion participation stats for a course; output is written to default location (same folder, auto-generated file name)
* `./manage.py lms export_discussion_participation <course_key> <file_name>` - exports entire discussion participation stats for a course; output is written chosen file (created if missing, overwritten if exists)
* `./manage.py lms export_discussion_participation <course_key> --thread-type=[discussion|question]` - exports discussion participation stats for a course for chosen thread type only.
* `./manage.py lms export_discussion_participation <course_key> --end-date=<iso8601 datetime>` - exports discussion participation stats for a course for threads/comments/replies created before specified date.
* `./manage.py lms export_discussion_participation <course_key> --end-date=<iso8601 datetime> --thread-type=[discussion|question]` - exports discussion participation stats for a course for threads/comments/replies created before specified date, including only threads of specified type

### CODE REVIEW

File: lms/djangoapps/django_comment_client/tests/test_utils.py [hunk 1]
```diff
@@ -644,3 +645,18 @@ def test_it(self):
...
+    @ddt.data(
+        ("normal.txt", "normal.txt"),
+        ("normal_with_alnum.csv", "normal_with_alnum.csv"),
+        ("normal_with_multiple_extensions.dot.csv", "normal_with_multiple_extensions.dot.csv"),
+        ("contains/slashes.html", "containsslashes.html"),
+        ("contains_symbols!@#$%^&*+=\|,.html", "contains_symbols.html"),
+        ("contains spaces.org", "contains_spaces.org"),
+    )
+    def test_format_filename(self, raw_filename, expected_output):
+        self.assertEqual(utils.format_filename(raw_filename), expected_output)
```
Comment (Kelketek) [inline lms/djangoapps/django_comment_client/tests/test_utils.py]: Newline missing at the end of file

File: lms/djangoapps/django_comment_client/management/commands/export_discussion_participation.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
+
+from courseware.courses import get_course
+from student.models import CourseEnrollment
+from lms.lib.comment_client.user import User
+import django_comment_client.utils as utils
+
+
+class _Fields:
```
Comment (Kelketek) [inline lms/djangoapps/django_comment_client/management/commands/export_discussion_participation.py]: As this class isn't used anywhere else, can we just place these constants in the root namespace of this module? Flat is better than nested.

File: lms/djangoapps/django_comment_client/utils.py [hunk 3]
```diff
@@ -436,3 +438,21 @@ def safe_content(content, course_id, is_staff=False):
             content[child_content_key] = safe_children
 
     return content
+
+
+def format_filename(s):
```
Comment (Kelketek) [inline lms/djangoapps/django_comment_client/utils.py]: Did you try using Django's slugify instead? Would it cause any problems here?

**Comment by Kelketek** [comment]
@e-kolpakov See my comments, but other than those, LGTM. Consider it my :+1: when you resolve them.

As for testing, check out the tests in django_comment_client/base/tests.py to see how the comment server was mocked previously.

**Comment by chrisndodge** [comment]
Hello @e-kolpakov and @Kelketek,

Interesting stuff. I just wanted to give a heads up that I'm working in django_comment_client (see my open running PR for Notifications integrations). So we should be careful that we don't end up stepping on each other, as we're both on tight deadlines :-)

FYI, @antoviaque 

- Chris

**Comment by mtyaka** [comment]
Don't have anything to add to @Kelketek's comments, LGTM :+1:

**Comment by antoviaque** [comment]
@chrisndodge Good to know, thanks for the warning. Are you changing anything that Eugeny is using here, ie, the format of the social API stats calls?

File: lms/djangoapps/django_comment_client/utils.py (hunk 3 above)
Comment (e-kolpakov) [inline lms/djangoapps/django_comment_client/utils.py]: @Kelketek it's not there in Django 1.4 :)

File: lms/djangoapps/django_comment_client/management/commands/export_discussion_participation.py (hunk 2 above)
Comment (e-kolpakov) [inline lms/djangoapps/django_comment_client/management/commands/export_discussion_participation.py]: @Kelketek on the other hand, having them in root namespace of the module would pollute the namespace. Plus, this class has semantic naming, so it's clear that these are some fields of something: namely fields of result record. So I believe these advantages outweight "Flat is better than nested" :)

**Pull request merged**